addons/hr_expense/models/account_payment.py

```diff
@@ -1,13 +1,25 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import models, _
+from odoo import models, fields, _
 from odoo.api import ondelete
 from odoo.exceptions import UserError
 
 
 class AccountPayment(models.Model):
     _inherit = "account.payment"
 
+    expense_sheet_id = fields.Many2one(related='move_id.expense_sheet_id')
+
+    def write(self, vals):
+        trigger_fields = {
+            'date', 'amount', 'payment_type', 'partner_type', 'payment_reference',
+            'currency_id', 'partner_id', 'destination_account_id', 'partner_bank_id', 'journal_id'
+            'ref', 'expense_sheet_id', 'payment_method_line_id'
+        }
+        if self.expense_sheet_id and any(field_name in trigger_fields for field_name in vals):
+            raise UserError(_("You cannot change the expense report linked to a payment."))
```

Comment:
error message doesn't make any sense: we're modifying a payment. What was wrong with the previous one?

Reply:
It's the same behavior as before, but it used to be done in the sync

Reply:
yeah but the message was 
raise UserError(_("You cannot do this modification since the payment is linked to an expense report."))

Reply:
updated